src/types.jl

```diff
@@ -42,7 +42,8 @@ Base.@kwdef struct Storage
     sd_table::DataFrames.DataFrame
 end
 
-Base.@kwdef struct DemandFlexibility
+Base.@kwdef mutable struct DemandFlexibility
```

Comment:
See my above comment about potentially reverting this struct back to being immutable.